src/plugins/vis_type_timelion/public/helpers/timelion_request_handler.ts

```diff
@@ -93,7 +96,15 @@ export function getTimelionRequestHandler({
 
     // parse the time range client side to make sure it behaves like other charts
     const timeRangeBounds = timefilter.calculateBounds(timeRange);
-    const sessionId = getDataSearch().session.getSessionId();
+    const isCurrentSession = () =>
```

Comment:
just interesting, why you defined `isCurrentSession` as a function and execute it each time.  

can we do something like: `const isCurrentSession = !!searchSessionId && searchSessionId === dataSearch.session.getSessionId();`

Reply:
we need to "recalculate" it in `finally` close because it could already be that when async call finishes `dataSearch.session.getSessionId();` has changed.
technically it won't be a problem to call `untrack` in that case either, but I am doing it for sanity, mostly to make it easier to debug the internal current session state by not calling `untrack` where we know it is not necessary